### PR TITLE
DCS-286 Extend the Responsible Officer contact page to fetch details from Delius.

### PR DESCRIPTION
**Pull request by bell-pa**
Also:
* Bump the Docker file base image to node:10.17-slim from node:10.15-slim
* Remove the probation-teams service from docker-compose.yaml because it isn't used

### CODE REVIEW

File: server/routes/contact.js [hunk 1]
```diff
@@ -1,21 +1,33 @@
...
+/**
+ *
+ * @param userAdminService
+ * @param {RoService} roService
+ * @returns {function(*): *}
+ */
+module.exports = (userAdminService, roService) => router => {
   router.get(
-    '/ro/',
+    '/:borkingId',
```
Comment (brightonsbox) [inline server/routes/contact.js]: borkingId?

Comment (andrewrlee) [reply server/routes/contact.js]: yeah.. maybe `associatedBookingId`?

File: types/probationTeams.d.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export interface ProbationTeamsClient {
```
Comment (andrewrlee) [inline types/probationTeams.d.ts]: this already exists in licences.d.ts

File: docker-compose.yml [hunk 3]
```diff
@@ -30,29 +30,6 @@ services:
       - GLOBAL_SEARCH_URL=http://localhost:3002/global-search
       - ENABLE_TEST_UTILS=true
 
-  probation-teams:
```
Comment (andrewrlee) [inline docker-compose.yml]: do we not want this any more?

File: types/probationTeams.d.ts (hunk 2 above)
Comment (andrewrlee) [reply types/probationTeams.d.ts]: oh no - I see you are just extracting it out.

Nothing wrong with that roContactDetailsService references the type from the probationteamclient directly - not sure if there is anything wrong with that either....

File: server/routes/contact.js [hunk 4]
```diff
@@ -1,21 +1,33 @@
...
+      // because 'bookingId' is treated specially - see standardRouter.js
+      const { borkingId } = req.params
+      const [ro] = unwrapResult(await roService.findResponsibleOfficer(borkingId, res.locals.token))
 
-  router.get(
-    '/ro/:deliusUserId',
-    asyncMiddleware(async (req, res) => {
-      const { deliusUserId } = req.params
-      const contact = await userAdminService.getRoUserByDeliusId(deliusUserId)
+      const contact = ro.deliusId ? await userAdminService.getRoUserByDeliusId(ro.deliusId) : undefined
```
Comment (andrewrlee) [inline server/routes/contact.js]: meh.. is `ro.deliusId && await userAdminService.getRoUserByDeliusId(ro.deliusId)` equivalent?

File: server/routes/contact.js [hunk 5]
```diff
@@ -1,21 +1,33 @@
...
-    '/ro/',
+    '/:borkingId',
     asyncMiddleware(async (req, res) => {
-      const roUsers = await userAdminService.getRoUsers()
-      return res.render('contact/roList', { roUsers })
-    })
-  )
+      // because 'bookingId' is treated specially - see standardRouter.js
+      const { borkingId } = req.params
+      const [ro] = unwrapResult(await roService.findResponsibleOfficer(borkingId, res.locals.token))
```
Comment (andrewrlee) [inline server/routes/contact.js]: `ro` would be undefined if an error is present?

File: server/routes/contact.js (hunk 4 above)
Comment (bell-pa) [reply server/routes/contact.js]: You're right. I've added a test, fixed this - and an equivalent bug lower down.

File: server/routes/contact.js (hunk 5 above)
Comment (bell-pa) [reply server/routes/contact.js]: yes. Fixed + test.

File: docker-compose.yml (hunk 3 above)
Comment (bell-pa) [reply docker-compose.yml]: discussed

**Pull request merged**